### PR TITLE
Host images with assets.scratch.mit.edu on the forums

### PR DESCRIPTION
**Pull request by veloce-vasta**
⬆⬆⬆⬆

Also credits to @TheGlassPenguin on Scratch: https://scratch.mit.edu/discuss/topic/572952/?page=1#post-6794037

### CODE REVIEW

**Comment by rgantzos** [comment]
This looks like a good idea, but what would it be used for?

**Comment by veloce-vasta** [comment]
> This looks like a good idea, but what would it be used for?

To host images without an external image hosting website, like cubeupload. All you would need to do is click the upload image button, and select a file, and it will insert the BB code.

image
Click on the img button
image
Select file
image
It inserts BB code
image
And works!

**Comment by rgantzos** [comment]
> > This looks like a good idea, but what would it be used for?
> 
> To host images without an external image hosting website, like cubeupload. All you would need to do is click the upload image button, and select a file, and it will insert the BB code.
> 
> image Click on the img button image Select file image It inserts BB code image And works!

Where did those screenshots come from? There doesn't seem to be any code in the js file, so I'm just curious.

**Comment by veloce-vasta** [comment]
Oh yeah, I used a bookmarklet for that, but rn I'm trying to find the ID of the button and list that contains the tools (image, bold, underline, etc.) Can you get it from inspect so that I know how to add an event listener to the button? I can't find it in the source code.

**Comment by veloce-vasta** [comment]
Also know that cubeupload and imageshack are blocked from school chromebook users.

**Comment by veloce-vasta** [comment]
Nvm, 
`javascript:(function()%7B(function() %7Bvar x %3D document.createElement("script")%3Bx.src %3D "https%3A%2F%2Fcdn.jsdelivr.net%2Fgh%2FSnowLord7%2Fdevconsole%40master%2Fmain.js"%3Bx.onload %3D alert("Loaded Developer Console!")%3Bdocument.head.appendChild(x)%3B%7D)()%7D)()`

Found ctrl !!! The picture button has class of `markItUpButton5`, so I can now make the feat.!

**Comment by veloce-vasta** [comment]
Ok

**Comment by rgantzos** [comment]
We have a new addToSpace API that might be helpful for that.

**Comment by veloce-vasta** [comment]
> We have a new addToSpace API that might be helpful for that.

Where?

It works, but for some reason the image isn't being displayed, but the alt text is. (I have good wifi)

**Comment by veloce-vasta** [comment]
Also, may I add a feature to make scroll-able signatures? It'll be real easy.

**Comment by veloce-vasta** [comment]
Done!

File: features/host-images-in-forums-with-assets.js
Comment (rgantzos) [inline features/host-images-in-forums-with-assets.js:9]: Could you please make it so that the cursor = pointer? That way it acts like the rest of the buttons.

File: features/features.json
Comment (rgantzos) [inline features/features.json:4]: The code all seems to work fine, but the title is too long, the description is too short, and too much information is given. Maybe a title such as "Direct image uploads in the forums" or something. For the description, keep in mind that these image uploads DO work for forum signatures.

File: features/features.json
Comment (rgantzos) [inline features/features.json:13]: Maybe add the "Recommended" tag?

File: features/features.json
Comment (rgantzos) [inline features/features.json:4]: Maybe a description like "Allows you to easily upload images to the Scratch forums for posts and signatures without having to use third party image uploading services."

**Review by rgantzos — CHANGES_REQUESTED** [review]

File: features/features.json
Comment (rgantzos) [reply features/features.json:4]: Just because we don't need to specify that it works for Chromebook users, and we also don't need to specify the image hosting platform.

File: features/features.json
Comment (rgantzos) [inline features/features.json:13]: The "Forums" tag doesn't do anything. We just need "New" and "Recommended".

**Comment by rgantzos** [comment]
Also, are you still able to do #258?

File: features/features.json
Comment (veloce-vasta) [reply features/features.json]: done

**Comment by rgantzos** [comment]
Looks good! I'll get one more review before merging, then I'll add your name to the contributors page.

**Comment by rgantzos** [comment]
> Ok (accidentally removed mistysloth as reviewer, never know that button did that, sorry)

It's fine lol, someone else is reviewing currently.

**Review by rgantzos — APPROVED** [review]

**Comment by rgantzos** [comment]
> > Also, are you still able to do #258?
> > Just curious because I'm adding it to the v2.21.0 milestone.
> 
> Sure! I'll try to hurry it up, it might be done by the 8th, is that fine with you?

That sounds good, thanks!

**Pull request merged**
